### PR TITLE
Fix bug in prefix enforcement when no subpackages are present

### PR DESCRIPTION
**Pull request by aadi-stripe**
A bug in package prefix enforcement (see attached failing test) prevents us from catching certain mis-packaged names when no sub-packages are present. This PR fixes that bug.

### Motivation

Bugfix.

### Test plan

See included automated tests. Tested on Stripe codebase.

### CODE REVIEW

**Comment by aadi-stripe** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_LtCKn1RfaFB3jZ
→ https://go/builds/bui_LtCK6yp8247Jk3

**Comment by aadi-stripe** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_LtD5dkt5lAu6uN
→ https://go/builds/bui_LtD5K8LWRK5srU

File: core/packages/PackageInfo.cc [hunk 1]
```diff
@@ -25,6 +25,10 @@ bool PackageInfo::lexCmp(const std::vector<core::NameRef> &lhs, const std::vecto
                                         [](NameRef a, NameRef b) -> bool { return a.rawId() < b.rawId(); });
 }
 
+bool PackageInfo::isEqual(const std::vector<core::NameRef> &lhs, const std::vector<core::NameRef> &rhs) {
+    return (lhs.size() == rhs.size() && std::equal(lhs.begin(), lhs.end(), rhs.begin(), rhs.end()));
+}
```
Comment (ngroman) [inline core/packages/PackageInfo.cc:30]: For future reference, you can compare &vecs with `==`. The std lib is smart enough to include the size check first.

**Review by ngroman — CHANGES_REQUESTED** [review]
Good catch on the bug. I was trying to get very cute. You don't actually need the full equality though, just a length check: I tested this in https://github.com/sorbet/sorbet/commit/2d44c08cf8bc235d6c69017c92add4877e5ea488

**Comment by aadi-stripe** [comment]
@ngroman Fixed, thanks!

**Pull request merged**